### DISCUSSION TITLE
openamp remoteproc driver

### DISCUSSION THREAD

**Discussion by mausys** [zephyrproject-rtos/zephyr#80229] [Ideas]
2 weeks ago I opened a pull request draft #79605, which implements a remoteproc driver as a counterpart to the linux remoteproc framework. The idea originally started when I tried to add carveouts to lib/open-amp/resource_table.c and realized, it would be nice to configure the carveouts in the devicetree. So far, I didn't got any response to the PR. Maybe because it is only a draft or because it is a new driver category. Maybe here is a more appropriate place to discuss this idea. The remoteproc driver is still work in progress, but any feedback is appreciated.

**Comment by vaishnavachath** [comment]
Nice work @mausys , I believe this will help reduce a lot of duplicate code in applications and make use of public APIs. I had a PoC for a remoteproc controller/master subsystem and driver for TI K3 devices, the intent was to make it easy to control remotecores (MCU/DSP .etc) from an application processor running Zephyr like TI AM62X. Once ready you may need to mark the PR ready for review and present in a dev-review meeting for wider feedback. Also will try to test this on our platforms and post the results on your PR.

**Comment by JarmouniA** [comment]
> 2 weeks ago I opened a pull request draft #79605

@mausys A draft PR generally means "not ready for review", so unless you tag people explicitly there will likely be no feedback until the PR is marked as ready and reviewers are assigned by the Zephyr Bot.
